src/containerapp/azext_containerapp/_utils.py

```diff
@@ -2,21 +2,218 @@
 # Copyright (c) Microsoft Corporation. All rights reserved.
 # Licensed under the MIT License. See License.txt in the project root for license information.
 # --------------------------------------------------------------------------------------------
-# pylint: disable=line-too-long, consider-using-f-string, no-else-return, duplicate-string-formatting-argument
+# pylint: disable=line-too-long, consider-using-f-string, no-else-return, duplicate-string-formatting-argument, expression-not-assigned, too-many-locals, logging-fstring-interpolation
+
+import time
+import json
+import platform
 
 from urllib.parse import urlparse
-from azure.cli.command_modules.appservice.custom import (_get_acr_cred)
-from azure.cli.core.azclierror import (ValidationError, RequiredArgumentMissingError)
+from datetime import datetime
+from dateutil.relativedelta import relativedelta
+from azure.cli.core.azclierror import (ValidationError, RequiredArgumentMissingError, CLIInternalError,
+                                       ResourceNotFoundError, ArgumentUsageError)
 from azure.cli.core.commands.client_factory import get_subscription_id
 from knack.log import get_logger
-from msrestazure.tools import parse_resource_id
+from msrestazure.tools import parse_resource_id, is_valid_resource_id, resource_id
 
 from ._clients import ContainerAppClient
 from ._client_factory import handle_raw_exception, providers_client_factory, cf_resource_groups, log_analytics_client_factory, log_analytics_shared_key_client_factory
 
 logger = get_logger(__name__)
 
 
+# original implementation at azure.cli.command_modules.role.custom.create_service_principal_for_rbac
+# reimplemented to remove unnecessary warning statements
```

Comment:
> \# logger.warning("Creating '%s' role assignment under scope '%s'", role, scope) 
> \# retry till server replication is done

I'm not sure whether it's reasonable to reimplement the whole `create_service_principal_for_rbac` method just to remove unnecessary warning statements. It has a disadvantage that it will cause the same code duplication and increase the maintenance cost. @jiasli Do you have any suggestions for this?

Reply:
To be more specific, I reimplemented the method to eliminate a warning statement that said: 
```
The output includes credentials that you must protect. 
Be sure that you do not include these credentials in your code or check the credentials into your source control.
For more information, see https://aka.ms/azadsp-cli
```

In this case, the warning is incorrect because these service principal credentials are not outputted (they are passed to the containerapp GitHub action API)

Reply:
I agree with @zhoxing-ms. Vendoring `create_service_principal_for_rbac` is a bad idea. `create_service_principal_for_rbac` is updated frequently due to design change or security requirements. Once vendored, you also need to redo Microsoft Graph migration (https://github.com/Azure/azure-cli/issues/22174). For example, `credential_description` and `password` are removed from `create_application`.

If you want the warning not to be shown, I can add a switch.